### DISCUSSION TITLE
ERROR: nvidia-container-toolkit-1.0.5-r0 do_fetch: Fetcher failure: Unable to find revision 60f165ad6901f85b0c3acbf7ce2c66cd759c4fb8 in branch master even from upstream

### DISCUSSION THREAD

**Discussion by danny1209** [OE4T/tegra-demo-distro#258] [Q&A] [answered]
Hi, 

I encounter this error since two days, could someone give some tips?
*******
WARNING: nvidia-container-toolkit-1.0.5-r0 do_fetch: Failed to fetch URL git://github.com/NVIDIA/container-toolkit;protocol=https, attempting MIRRORS if available
ERROR: nvidia-container-toolkit-1.0.5-r0 do_fetch: Fetcher failure: Unable to find revision **60f165ad6901f85b0c3acbf7ce2c66cd759c4fb8** in branch master even from upstream
ERROR: nvidia-container-toolkit-1.0.5-r0 do_fetch: Bitbake Fetcher Error: FetchError('Unable to fetch URL from any source.', 'git://github.com/NVIDIA/container-toolkit;protocol=https')
ERROR: Logfile of failure stored in: /home/nian/Heron/linux_oe4t_tegra_demo_distro/build/tmp/work/aarch64-oe4t-linux/nvidia-container-toolkit/1.0.5-r0/temp/log.do_fetch.4879
ERROR: Task (/home/nian/Heron/linux_oe4t_tegra_demo_distro/layers/meta-tegra/external/virtualization-layer/recipes-containers/nvidia-container-toolkit/nvidia-container-toolkit_1.0.5.bb:do_fetch) failed with exit code '1'
NOTE: Tasks Summary: Attempted 5571 tasks of which 5501 didn't need to be rerun and 1 failed.

Summary: 1 task failed:
  /home/nian/Heron/linux_oe4t_tegra_demo_distro/layers/meta-tegra/external/virtualization-layer/recipes-containers/nvidia-container-toolkit/nvidia-container-toolkit_1.0.5.bb:do_fetch
Summary: There was 1 WARNING message shown.
Summary: There were 2 ERROR messages shown, returning a non-zero exit code.
*******

BR
Danny

**Comment by ichergui** [comment]
Hey @danny1209 
Are you sure about you network ? 
I tried to fetch `nvidia-container-toolkit` manually. It is working properly.
```
➜  /tmp git clone https://github.com/NVIDIA/nvidia-container-toolkit.git   
Cloning into 'nvidia-container-toolkit'...
remote: Enumerating objects: 9155, done.
remote: Counting objects: 100% (1543/1543), done.
remote: Compressing objects: 100% (591/591), done.
remote: Total 9155 (delta 1012), reused 1333 (delta 905), pack-reused 7612
Receiving objects: 100% (9155/9155), 8.22 MiB | 4.59 MiB/s, done.
Resolving deltas: 100% (5113/5113), done.
➜  /tmp cd nvidia-container-toolkit 
➜  nvidia-container-toolkit git:(main)  (main)git checkout 60f165ad6901f85b0c3acbf7ce2c66cd759c4fb8              
Note: switching to '60f165ad6901f85b0c3acbf7ce2c66cd759c4fb8'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at 60f165a Merge branch 'toolkit_no_pivot' into 'master'
➜  nvidia-container-toolkit git:(60f165a)  (v1.1.0~5)

```

**Reply by danny1209** [reply]
Hi @ichergui, 

thanks for the reply! 
manually fetching works also perfektly for me, but not in bitbake.

and the source description in recipe confuses me also, from which repo will it be fetched? 
![image](https://github.com/OE4T/tegra-demo-distro/assets/46848895/dcd24358-f0a3-4223-b6d2-13c282636c07)

**Comment by ichergui** [comment]
Which branch you are using ? 
I think you are using `dunfell-l4t-r32.5.0`. Could you please confirm that ?

**Reply by danny1209** [reply]
dunfell

**Reply by ichergui** [reply]
> dunfell

Are you sure ? 
I think the `dunfell` is looking good to me.

Kindly see the link below: 

https://github.com/OE4T/meta-tegra/blob/dunfell/external/virtualization-layer/recipes-containers/nvidia-container-toolkit/nvidia-container-toolkit_1.0.5.bb

**Comment by danny1209** [comment]
I see now
![image](https://github.com/OE4T/tegra-demo-distro/assets/46848895/0fcdc0f4-c7f8-4298-9ba8-992dc0b75f93)

**Reply by ichergui** [reply]
Please make sure you have an updated branch. 
In this case try to `clean` then `build` again `nvidia-container-toolkit` recipe.

**Reply by danny1209** [reply, marked as answer]
I imported the dunfell branch in our own repo in Jan this year, this update was made on Mar 17 from your site. 
Somehow my  repo does not follow up your update automatically, therefore I missed this update and it´s not working
